### PR TITLE
feat: update dependencies and rust version

### PR DESCRIPTION
**Pull request by somehowchris**
There are 3 parts to this pull requeest:
- Update to rust 2021
- Update and tag dependencies
- Freshen up the `.gitignore` file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
+
+### VisualStudioCode ###
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+!.vscode/*.code-snippets
```
Comment (najtin) [inline .gitignore:27]: To me these exceptions suggest that these files are supposed to be tracked by git. Is that what you are implying here?

File: Cargo.toml [hunk 2]
```diff
@@ -17,28 +17,28 @@ json = ["serde_json"]
 yaml = ["serde_yaml"]
 parse-value = ["pear"]
 test = ["tempfile", "parking_lot"]
-# toml = ["toml"]
+# toml = ["toml"] -> usable once we can select the dependency as dep:toml
```
Comment (najtin) [inline Cargo.toml:20]: Could elaborate on that? I don't quit understand.

Comment (najtin) [inline Cargo.toml:20]: Could elaborate on that? I don't quit understand.

File: .gitignore (hunk 1 above)
Comment (somehowchris) [reply .gitignore:27]: Yes

File: Cargo.toml (hunk 2 above)
Comment (somehowchris) [reply Cargo.toml:20]: One can not have a depending crate and feature with the same name, the comment was already there so I extended it. There is a feature in nightly where that will be allowed. You could then have a feature `toml` as well as a crate `toml`, declare the feature and use `dep:toml` to declare the dependency of that feature on the crate

File: .gitignore (hunk 1 above)
Comment (najtin) [reply .gitignore:27]: Why do you think these should be tracked?

Comment (somehowchris) [reply .gitignore:27]: Well as the file says, I used [gitignore.io](https://www.toptal.com/developers/gitignore) which has been a lovel tool so far.

Indeed these should be tracked, as a vscode setup with tasks, launch options,... should be portable

Comment (najtin) [reply .gitignore:27]: Okay, even firefox tracks some of these files. https://hg.mozilla.org/mozilla-central/file/tip/.vscode Though we have to be careful not to add any clutter or local settings.

**Review by najtin — APPROVED** [review]
all my concerns/questions where eradicated/answered in the comments

**Pull request closed without merging**